File: Ass-1.py
import sys

def solve():
    # We use sys.stdin.read() to read all input at once because 
    # reading line-by-line with N=200,000 can be too slow in Python.
    input_data = sys.stdin.read().split()
    
    if not input_data:
        return

    # This creates an iterator to walk through our input data one by one
    iterator = iter(input_data)
    
    try:
        # Read T (Number of test cases)
        T_str = next(iterator, None)
        if not T_str: return
        T = int(T_str)
        
        for _ in range(T):
            # Read N (Number of meetings)
            N = int(next(iterator))
            meetings = []
            
            # Read the Start and End times
            for _ in range(N):
                s = int(next(iterator))
                e = int(next(iterator))
                meetings.append([s, e])
            
            # --- THE SORTING STEP ---
            # Sort by End Time (index 1)
            # key=lambda x: x[1] is the same as the Java Comparator
            meetings.sort(key=lambda x: x[1])
            
            # --- GREEDY LOOP ---
            count = 1
            last_end_time = meetings[0][1]
            
            for i in range(1, N):
                # Check if next meeting starts after (or when) previous ended
                if meetings[i][0] >= last_end_time:
                    count += 1
                    last_end_time = meetings[i][1]
            
            print(count)
            
    except StopIteration:
        pass

if __name__ == "__main__":
    solve()

File: Ass_1.java
import java.util.*;

public class Ass_1
 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        if (sc.hasNextInt()) {
            int T = sc.nextInt(); // Number of test cases
            
            while (T-- > 0) {
                int N = sc.nextInt(); // Number of meetings
                
                // Create a 2D array to store Start and End times
                // meetings[i][0] = Start Time
                // meetings[i][1] = End Time
                int[][] meetings = new int[N][2];
                
                for (int i = 0; i < N; i++) {
                    meetings[i][0] = sc.nextInt();
                    meetings[i][1] = sc.nextInt();
                }
                
                // Sort the 2D array based on End Time (column 1)
                Arrays.sort(meetings, (a, b) -> Integer.compare(a[1], b[1]));
                
                int count = 1; // We always pick the first meeting
                int lastEndTime = meetings[0][1];
                
                // Iterate through the rest of the meetings
                for (int i = 1; i < N; i++) {
                    // If current meeting starts after (or when) previous ended
                    if (meetings[i][0] >= lastEndTime) {
                        count++;
                        lastEndTime = meetings[i][1]; // Update end time
                    }
                }
                
                System.out.println(count);
            }
        }
    }
}